plugins/flytekit-snowflake/flytekitplugins/snowflake/agent.py

```diff
@@ -0,0 +1,120 @@
+import datetime
+import json
+from dataclasses import asdict, dataclass
+from typing import Dict, Optional
+
+import snowflake.connector
+from snowflake.connector import ProgrammingError
+
+import grpc
+from flyteidl.admin.agent_pb2 import (
+    PERMANENT_FAILURE,
+    SUCCEEDED,
+    CreateTaskResponse,
+    DeleteTaskResponse,
+    GetTaskResponse,
+    Resource,
+)
+
+from flytekit import FlyteContextManager, StructuredDataset, logger
+from flytekit.core.type_engine import TypeEngine
+from flytekit.extend.backend.base_agent import AgentBase, AgentRegistry, convert_to_flyte_state
+from flytekit.models import literals
+from flytekit.models.literals import LiteralMap
+from flytekit.models.task import TaskTemplate
+from flytekit.models.types import LiteralType, StructuredDatasetType
+
+pythonTypeToBigQueryType: Dict[type, str] = {
+    # https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#data_type_sizes
+    list: "ARRAY",
+    bool: "BOOL",
+    bytes: "BYTES",
+    datetime.datetime: "DATETIME",
+    float: "FLOAT64",
+    int: "INT64",
+    str: "STRING",
+}
+
+
+@dataclass
+class Metadata:
+    query_id: int
+
+
+class SnowflakeAgent(AgentBase):
+    def __init__(self):
+        super().__init__(task_type="snowflake")
+
+    def create(
+        self,
+        context: grpc.ServicerContext,
+        output_prefix: str,
+        task_template: TaskTemplate,
+        inputs: Optional[LiteralMap] = None,
+    ) -> CreateTaskResponse:
+        params = None
+        if inputs:
+            ctx = FlyteContextManager.current_context()
+            python_interface_inputs = {
+                name: TypeEngine.guess_python_type(lt.type) for name, lt in task_template.interface.inputs.items()
+            }
+            native_inputs = TypeEngine.literal_map_to_kwargs(ctx, inputs, python_interface_inputs)
+            logger.info(f"Create Snowflake params with inputs: {native_inputs}")
+            params = native_inputs
+
+        config = task_template.config
+        self.conn = snowflake.connector.connect(
+            user=config["user"],
+            password=config["password"],
+            account=config["account"],
+            database=config["database"],
+            schema=config["schema"],
+            warehouse=config["warehouse"]
+        )
+
+        self.cs = self.conn.cursor()
+        self.cs.execute_async(task_template.sql.statement, params=params)
```

Comment:
We should render the query. https://github.com/flyteorg/flyteplugins/blob/03f5b5c709277f49b04b65b591c469695defd8d4/go/tasks/plugins/webapi/snowflake/plugin.go#L88-L95